prov/shm/src/smr_ep.c

```diff
@@ -386,23 +371,34 @@ int smr_iov_mmap_copy_in(struct smr_ep *ep, struct smr_region *peer_smr,
 		num = ofi_copy_from_iov(mapped_ptr, total_len, iov, count, 0);
 		if (num != total_len) {
 			FI_WARN(&smr_prov, FI_LOG_EP_CTRL, "copy from iov error\n");
-			goto err2;
+			goto err3;
 		}
 		munmap(mapped_ptr, total_len);
+		pend->map_ptr = NULL;
 	} else {
-		*shm_ptr = mapped_ptr;
+		pend->map_ptr = mapped_ptr;
 	}
-
 	close(fd);
+
+	smr_generic_format(cmd, peer_id, op, tag, 0, 0, data, op_flags);
+	cmd->msg.hdr.op_src = smr_src_mmap;
+	cmd->msg.hdr.msg_id = msg_id;
+	cmd->msg.hdr.src_data = (uintptr_t) ((char **) resp - (char **) ep->region);
+	cmd->msg.hdr.size = total_len;
+	smr_post_pend_resp(pend, cmd, context, iov, count, resp);
+	pend->map_name = map_name;
+
 	return 0;
 
+err3:
+	munmap(mapped_ptr, total_len);
 err2:
-	dlist_remove(&(*map_name)->entry);
-	free(*map_name);
+	dlist_remove(&(map_name)->entry);
+	free(map_name);
+err1:
 	shm_unlink(shm_name);
 	close(fd);
-err1:
-	return -errno;
+	return ret;
```

Comment:
Please don't have patches modify other patches in the series.  Update the original patch directly.

Reply:
Updated.